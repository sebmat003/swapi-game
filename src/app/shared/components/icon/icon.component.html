<svg [attr.width]="width" [attr.height]="height">
  <use [attr.xlink:href]="url + icon"></use>
</svg>
