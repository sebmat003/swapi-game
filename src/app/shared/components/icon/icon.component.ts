import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'app-icon',
  templateUrl: './icon.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class IconComponent {
  @Input() url = 'assets/icons/icons.svg#';
  @Input({ required: true }) icon!: string;
  @Input() width = 48;
  @Input() height = 48;
}
