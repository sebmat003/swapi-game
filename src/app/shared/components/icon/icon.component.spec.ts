import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { IconComponent } from "./icon.component";

describe('IconComponent', () => {
  let spectator: Spectator<IconComponent>;
  const createComponent = createComponentFactory({
    component: IconComponent,
    detectChanges: false
  });

  beforeEach(() => spectator = createComponent())

  it('Should match snapshot', () => {
    spectator.setInput('icon', 'person');

    expect(spectator.fixture).toMatchSnapshot();
  });

  it('Should set provided width and height of icon', () => {
    const width = 100;
    const height = 100;

    spectator.setInput('width', width);
    spectator.setInput('height', height);

    expect(spectator.query('svg')?.getAttribute('width')).toEqual(width.toString());
    expect(spectator.query('svg')?.getAttribute('height')).toEqual(height.toString());
  })
});
