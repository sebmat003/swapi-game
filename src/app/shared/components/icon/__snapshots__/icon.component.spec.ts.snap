// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent Should match snapshot 1`] = `
<app-icon
  height={[Function Number]}
  icon={[Function String]}
  url={[Function String]}
  width={[Function Number]}
>
  <svg
    height="48"
    width="48"
  >
    <use
      xlink:href="assets/icons/icons.svg#person"
    />
  </svg>
</app-icon>
`;
